src/wallet/reconstruct_seed_flow.c

```diff
@@ -0,0 +1,304 @@
+// TODO: Add these pointers in a common header file
+extern char *ALPHABET;
+extern char *ALPHA_NUMERIC;
+extern char *NUMBERS;
+extern char *PASSPHRASE;
+
+// TODO: Remove usage of global variables
+extern Wallet_shamir_data wallet_shamir_data;
+extern Wallet_credential_data wallet_credential_data;
+extern Wallet wallet;
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef enum {
+  PASSPHRASE_INPUT = 1,
+  PASSPHRASE_CONFIRM,
+  PIN_INPUT,
+  TAP_CARD_FLOW,
+  RECONSTRUCT_SEED,
+  COMPLETED,
+  COMPLETED_WITH_ERRORS,
+  TIMED_OUT,
+  EARLY_EXIT,
+  EXIT,
+} reconstruct_state_e;
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+/**
+ * @brief This function handles different states of the reconstruct wallet seed
+ * flow
+ *
+ * @param state The current state of the flow which needs to be executed
+ * @param seed_out Pointer to buffer which needs to be populated with the wallet
+ * seed
+ * @return reconstruct_state_e The next state of the flow
+ */
+reconstruct_state_e reconstruct_seed_handler(reconstruct_state_e state,
+                                             uint8_t *seed_out);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+reconstruct_state_e reconstruct_seed_handler(reconstruct_state_e state,
+                                             uint8_t *seed_out) {
+  reconstruct_state_e next_state = EXIT;
+  switch (state) {
+    case PASSPHRASE_INPUT: {
+      if (!WALLET_IS_PASSPHRASE_SET(wallet.wallet_info)) {
+        next_state = PIN_INPUT;
+        break;
+      }
+
+      input_text_init(
+          PASSPHRASE, ui_text_enter_passphrase, 0, DATA_TYPE_PASSPHRASE, 64);
+      next_state =
+          get_state_on_input_scr(PASSPHRASE_CONFIRM, EARLY_EXIT, TIMED_OUT);
+      break;
+    }
+
+    case PASSPHRASE_CONFIRM: {
+      char display[65];
+      snprintf(
+          display, sizeof(display), "%s", flow_level.screen_input.input_text);
+      address_scr_init(ui_text_confirm_passphrase, display, false);
+      memzero(display, sizeof(display));
```

Comment:
We could remove the variable display here

Reply:
How can we do that? Passphrase is stored in that variable to display using `address_scr_init` in line #153

Reply:
Updated to use scroll page